runtime-watcher/internal/watchermetrics/godebug.go

```diff
@@ -0,0 +1,35 @@
+package watchermetrics
+
+import (
+	"strings"
+)
+
+// parseGodebugFipsMode parses the provided value to determine the FIPS mode.
+// The argument to this function should be a value of the GODEBUG environment variable.
+// We need to parse it manually because the provided fips140.Enabled() function doesn't distinguish
+// between "on", "debug" and "only" modes.
+func parseGodebugFipsMode(goDebugEnvVar string) string {
+	const off = "off"
+
+	result := off
+
+	if len(goDebugEnvVar) == 0 {
+		return result
+	}
+	for keyValuePair := range strings.SplitSeq(goDebugEnvVar, ",") {
```

Comment:
This uses `strings.SplitSeq` which returns an iterator. While it  s working, the more conventional approach would be: `strings.Split`.
I am okay with either , but perhaps you thought about memory efficiency targeting performance with large string? but we have this case?